### PR TITLE
Validation module for GEM pad digi clusters

### PR DESCRIPTION
**Pull request by dildick**
#### PR description:

This PR introduces a validation module for GEM pad digi clusters. It is based on the GEM pad digi validation module. I also cleaned up some of the outdated code. For instance, I replaced the "generic" digis with the actual GEM digis/pads throughout the code.

#### PR validation:

It tested this with WF 11634.0. The new cluster plots are presented in the DQM ROOT file. The first DQM plot shows the locations (in R-Z) of the strip digis in sample with 10 events. The second DQM plot shows the corresponding pads (OR of two strip digis). The third DQM plot show the clusters of neighboring pads.

Screen Shot 2019-10-24 at 5 47 22 PM

Screen Shot 2019-10-24 at 5 22 06 PM

Screen Shot 2019-10-24 at 5 20 48 PM

#### if this PR is a backport please specify the original PR:

N/A

@jshlee @tahuang1991

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

File: Validation/MuonGEMDigis/plugins/GEMCheckGeometry.cc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    temp_me->setBinLabel(7, "St2,La2_even", 2);
+    temp_me->setBinLabel(8, "St2,La2_odd", 2);
+    theStdPlots.insert(std::map<UInt_t, MonitorElement*>::value_type(name.Hash(), temp_me));
+  }
+
+  for (auto region : GEMGeometry_->regions())
+    for (auto station : region->stations())
+      for (auto ring : station->rings())
+        for (auto sch : ring->superChambers())
+          for (auto ch : sch->chambers())
```
Comment (jfernan2) [inline Validation/MuonGEMDigis/plugins/GEMCheckGeometry.cc]: Not using brackets it is considered error prone code IMHO, but I let the experts to decide.
Moreover, region, station, ring, sch and ch iterators are not used inside the loops apparently, so what's the purpouse of the loops?

**Comment by dildick** [comment]
Not sure what happened here. I'll look into it

**Comment by dildick** [comment]
I moved all DQM analyzers and producers to the plugins directory, ran `scram b`, `scram b code-checks` and `scram b code-format`.

**Comment by dildick** [comment]
@jfernan2 I added the missing brackets. Can you rerun the tests?

**Comment by jfernan2** [comment]
please test

**Comment by dildick** [comment]
@jfernan2 I replaced the GenericDigi in the GEM rechits matching module by the actual GEM hit objects in CMSSW. WF 11634 runs, except for a few warnings in `GEMRecHitTrackMatch` code. I'm still working on this.

**Comment by dildick** [comment]
@jfernan2 I'm closing this PR for now; upon further inspection, more cleanup is needed of the GEM validation code. I'll submit a new PR once that is done.

**Pull request closed without merging**